Docs: Add docs for variations in the block registration section (#20145)
WordPress/gutenberg: docs/designers-developers/developers/block-api/block-registration.md

If `example` is not defined, the preview will not be shown. So even if no-attributes are defined, setting a empty example object `example: {}` will trigger the preview to show.

#### variations (optional)

- **Type:** `Array`

> Similar to how the block's style variations can be declared, a block type can define block variations the user can pick from. The difference is that rather than changing only the visual appearance, this field provides a way to apply initial custom attributes and inner blocks at the time when a block is inserted. All variations by default will show up in the inserter in addition to the regular block type item. When the `isDefault` flag is set for any of the variations listed, then it overrides the regular block type item in the inserter instead.

Reviewer: ```suggestion
Similarly to how the block's style variations can be declared, a block type can define block variations that the user can pick from. The difference is that, rather than changing only the visual appearance, this field provides a way to apply initial custom attributes and inner blocks at the time when a block is inserted.

By default, all variations will show up in the Inserter in addition to the regular block type item. However, setting the `isDefault` flag for any of the variations listed will override the regular block type in the Inserter.
```

Other notes:

- It's good that you brought up style variations, because we should be pretty clear about what each one is.
- _"rather than changing only the visual appearance"_: You explain style variations in terms of meaning and function, which is good, instead of explaining the implementation detail (that it applies an HTML class for style targeting). In contrast, the explanation for block variations seems more based in the implementation (_"a way to apply initial custom attributes and […]"_). It might be better to say something like "block variations are specific arrangements of a block with preset attributes or children".

{% codetabs %}
{% ES5 %}

```js
variations: [
    {
		name: 'wordpress',
		isDefault: true,
		title: __( 'WordPress' ),
		description: __( 'Code is poetry!' ),
		icon: WordPressIcon,

> 		attributes: { service: 'wordpress' },

Reviewer: Should distinguish the required fields from the optional ones.

Author: Yes, I can list of fields with their descriptions to make it clear.

Author: Added in 797c78f.
